### PR TITLE
Add emitWorkflowTypeCountMetricsPinot in ESAnalyzer

### PR DESCRIPTION
**Pull request by bowenxia**
**What changed?**
Add emitWorkflowTypeCountMetricsPinot in ESAnalyzer, 
Add pinot aggregation query that has similar result as ES.
Update unit test accordingly.

**Why?**
To upgrade ESAnalayzer to be Visibility Analyzer

**How did you test it?**
Unit test

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/pinot/pinot_client.go [hunk 1]
```diff
@@ -70,6 +70,18 @@ func (c *PinotClient) Search(request *SearchRequest) (*SearchResponse, error) {
 	return c.getInternalListWorkflowExecutionsResponse(resp, request.Filter, token, request.ListRequest.PageSize, request.MaxResultWindow)
 }
 
+func (c *PinotClient) SearchAggr(request *SearchRequest) (AggrResponse, error) {
```
Comment (3vilhamster) [inline common/pinot/pinot_client.go:73]: this method is missing tests as far as I see

Comment (bowenxia) [reply common/pinot/pinot_client.go:73]: Added

File: service/worker/esanalyzer/domainWorkflowTypeCountWorkflow.go [hunk 2]
```diff
@@ -160,6 +165,80 @@ func (w *Workflow) emitWorkflowTypeCountMetrics(ctx context.Context) error {
...
+	return fmt.Sprintf(`
+SELECT WorkflowType, COUNT(*) AS count
+FROM %s
+WHERE DomainID = '%s'
+AND IsDeleted = false
+AND CloseStatus = -1
+AND StartTime > 0
+GROUP BY WorkflowType
+ORDER BY count
+LIMIT 10
```
Comment (shijiesheng) [inline service/worker/esanalyzer/domainWorkflowTypeCountWorkflow.go:183]: ES search returns at most top 10 aggregation results. I don't see ORDER by here in pinot query

**Review by shijiesheng — APPROVED** [review]

File: service/worker/esanalyzer/domainWorkflowTypeCountWorkflow.go [hunk 3]
```diff
@@ -160,6 +165,81 @@ func (w *Workflow) emitWorkflowTypeCountMetrics(ctx context.Context) error {
...
+	domain, err := w.analyzer.domainCache.GetDomain(domainName)
+	if err != nil {
+		return "", err
+	}
+	// exclude uninitialized workflow executions by checking whether record has start time field
+	// there's a "LIMIT 10" because in ES, Aggr clause by default returns the top 10 results
+	return fmt.Sprintf(`
+SELECT WorkflowType, COUNT(*) AS count
+FROM %s
+WHERE DomainID = '%s'
```
Comment (davidporter-id-au) [inline service/worker/esanalyzer/domainWorkflowTypeCountWorkflow.go:178]: nit: `%q`

Comment (bowenxia) [reply service/worker/esanalyzer/domainWorkflowTypeCountWorkflow.go:178]: `%q` here might not work because double quote doesn't work in Pinot

**Pull request merged**